README.md

```diff
@@ -1,58 +1,155 @@
+
+```sh
+docker run --rm manageiq/rpm_build:jansa-3 build --update-rpm-repo
+```
+
+## Building a custom release
+
+Sometimes it is necessary to build rpms with code from pull requests. This explains how to do that.
 
-2. Prepare configuration
+### Defining build parameters
+
+The [options.yml](blob/master/config/options.yml) points to the appropriate repos and tag/branches.
+
+A typical `options.yml` will only override the source and the refs. This example uses branch `feature1` from kbrock's github repo:
+
+```sh
+mkdir OPTIONS
+vi OPTIONS/options.yml
+```
+
+```yaml
+---
+product_name:      manageiq
+repos:
+  ref:             master
+  manageiq:
+    url:           https://github.com/kbrock/manageiq.git
+    ref:           feature1
+  manageiq_appliance:
+    url:           https://github.com/kbrock/manageiq-appliance.git
+    ref:           feature1
+```
+
+The option file is brought into docker by mounting the directory with `-v $(pwd)/OPTIONS:/root/OPTIONS`
+
+Alternatively, a static directory may used like `/opt/OPTIONS/options.yml`.
+
+### Using a custom NPM registry
+
+Any of the keys in [options.yml](blob/master/config/options.yml) can be overridden.
 
-   - Create a local directory with an `options.yml` with the configuration changes
-     you want to make, then when running the container image, mount that directory
-     with `-v <dir>:/root/OPTIONS`.
    - If overriding the NPM registry, set the `npm_registry` key in the `options.yml`.
    - If building RPMs in Copr,
      - set the `rpm.repo_name` key in the `options.yml`.
      - run the container image with `-v <copr token file>:/root/.config/copr`.
    - If updating the RPM repo it would be helpful to attach a volume to hold the RPM cache with `-v <dir>:/root/rpm_cache`.
      Any RPMs not in the cache will be downloaded to the cache first.
 
-3. Start the container image
+### Artifacts
 
-   `docker run -it <image>`
+The purpose of this process is to build code from `manageiq-core`, `manageiq-gemset`, and `manageiq-appliance` into 2 sets of artifacts:
 
-4. Run the script
+- `/root/BUILD/rpm_spec/*.tar.gz`
+- `/root/BUILD/rpms/x86_64/manageis-*.rpm`
 
-   - Run `bin/build.rb`, optionally overriding git ref (branch or tag) and release type
+Depending upon your usecase, there are a few ways to obtain these artifacts:
 
-     ```
-     bin/build.rb --git-ref jansa --update-rpm-repo
-     ```
-     ```
-     bin/build.rb --build-type release --git-ref jansa-1-alpha1 --update-rpm-repo
-     ```
+- Have the container put the artifacts into a mounted volume.
+- Manually copy the artifacts to a mounted volume.
+- Extracting the artifacts from the docker container with a docker command.
 
-Alternatively, the build can be started directly by passing `build` to the
-`docker run` command:
+### Having the container build files into a mounted volume
 
+Since the files are not uploaded, they need to be retrieved out of the repo before the container exits.
```

Comment:
```suggestion
Since the files are not uploaded, they need to be retrieved from the repo before the container exits.
```

Reply:
IIRC, all of the files still exist on the mounted directory after the container exits, correct?

Reply:
once the primary process exists, you can not access the container anymore.

Reply:
I think I'm wrong here. I'll revisit and fix.

Since I am wrong, this greatly simplifies things. (thanks to jason't comments down below)